File: app/src/main/java/com/asf/wallet/di/GasSettingsModule.java
package com.asf.wallet.di;

import com.asf.wallet.interact.FindDefaultNetworkInteract;
import com.asf.wallet.repository.EthereumNetworkRepositoryType;
import com.asf.wallet.viewmodel.GasSettingsViewModelFactory;
import dagger.Module;
import dagger.Provides;

@Module public class GasSettingsModule {

  @Provides public GasSettingsViewModelFactory provideGasSettingsViewModelFactory(
      FindDefaultNetworkInteract findDefaultNetworkInteract) {
    return new GasSettingsViewModelFactory(findDefaultNetworkInteract);
  }

  @Provides FindDefaultNetworkInteract provideFindDefaultNetworkInteract(
      EthereumNetworkRepositoryType ethereumNetworkRepositoryType) {
    return new FindDefaultNetworkInteract(ethereumNetworkRepositoryType);
  }
}

File: app/src/main/java/com/asfoundation/wallet/viewmodel/GasSettingsModule.kt
package com.asfoundation.wallet.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.asfoundation.wallet.ui.GasSettingsInteractor
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@InstallIn(ActivityComponent::class)
@Module
class GasSettingsModule {
  @Provides
  fun provideGasSettingsViewModelFactory(gasSettingsInteractor: GasSettingsInteractor) =
    GasSettingsViewModelFactory(gasSettingsInteractor)
}

class GasSettingsViewModelFactory(private val gasSettingsInteractor: GasSettingsInteractor) :
  ViewModelProvider.Factory {

  override fun <T : ViewModel> create(modelClass: Class<T>): T {
    return GasSettingsViewModel(gasSettingsInteractor) as T
  }
}